### PR TITLE
Merge to Development PR

### PR DESCRIPTION
**Pull request by wathigo**
## Tasks Completed
1. Creation of player, game board, and ship factories.
2. Implementation of the tests for the factories.
3. Implementation of dom tasks.

Please review our PR. Thank you.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "jest"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/samgaco/battleship.git",
+    "dev": "webpack --mode development",
+    "build": "webpack --mode production",
+    "start": "webpack-dev-server --mode development --open"
```
Comment (azdrenymeri) [inline package.json:10]: Your run commands should go under the `scripts`

File: src/factories/ship.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  hits: 0,
+  hit() {
+    this.isSunk() ? undefined : this.hits += 1;
+  },
+  isSunk() {
+    return this.hits >= this.length;
+  },
+}));
+
+module.exports = newShip;
```
Comment (azdrenymeri) [inline src/factories/ship.js]: Use `export` instead of `module.exports`

File: src/factories/player.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    movesdone.push(randMove);
+    return coorMaps[randMove];
+  };
+
+  return {
+    aiPlay, PosTaken, turn, movesdone, coorMaps,
+  };
+});
+
+module.exports = player;
```
Comment (azdrenymeri) [inline src/factories/player.js]: Same here use(the ES6) `export` instead

File: src/factories/gameBoard.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    return this.ships.length - this.shipsSunk();
+  },
+
+  gameStop() {
+    return this.shipsSunk() === this.ships.length;
+  },
+}));
+
+
+module.exports = gameBoard;
```
Comment (azdrenymeri) [inline src/factories/gameBoard.js]: Same as `ship.js` and `player.js` use `export`

File: src/factories/player.js (hunk 3 above)
Comment (azdrenymeri) [inline src/factories/player.js]: Same here use `export`

File: src/tests/gameBoard.test.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
+const gameBoard = require('../factories/gameBoard');
+const newShip = require('../factories/ship');
```
Comment (azdrenymeri) [inline src/tests/gameBoard.test.js]: Use  `import` instead of `require`

File: src/tests/ship.test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const newShip = require('../factories/ship');
```
Comment (azdrenymeri) [inline src/tests/ship.test.js]: Same here use `import` instead of `require`

File: src/factories/gameBoard.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        }
+      }
+    });
+    if (found === false) {
+      this.table[coord[0]][coord[1]] = 'M';
+    }
+    return this.table[coord[0]][coord[1]];
+  },
+
+  shipsSunk() {
```
Comment (azdrenymeri) [inline src/factories/gameBoard.js:58]: A more shorter and concise way is using high-order functions
`return this.ships.filter(ship => ship.isSunk() === true).length`

File: src/factories/player.js (hunk 3 above)
Comment (azdrenymeri) [inline src/factories/player.js]: Same here

**Review by azdrenymeri — CHANGES_REQUESTED** [review]
## Hello @wathigo  and @samgaco 👋 
### You did a great job 👍 
You have been using `module.exports` and `require` in some of your `js` files.
Kindly change that and use the ES6 `import/export`.
To be able to use `import/export` with `Jest` you need to install babel which you already did
but `.babelrc` file is missing on your project so kindly add it and check this [Link](https://stackoverflow.com/a/49656707/6539808) for further instructions.
### Kindly implement the changes and add it for a re-review

**Review by JamesOkunlade — APPROVED** [review]
# Approved 🙂 

### Great job on this project and nice job implementing the corrections requested by the previous reviewer.

It's 👍 for me. Go ahead and merge. 

Good luck with your next project.

**James Okunlade**
Happy coding! 😎

**Pull request merged**